Sources/MongoSwift/ChangeStream.swift

```diff
@@ -5,20 +5,26 @@ import mongoc
 public struct ResumeToken: Codable, Equatable {
     private let resumeToken: Document
 
-    internal init(resumeToken: Document) {
+    internal init(_ resumeToken: Document) {
         self.resumeToken = resumeToken
     }
 
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.singleValueContainer()
+        try container.encode(self.resumeToken)
+    }
+
     public init(from decoder: Decoder) throws {
-        self.resumeToken = try Document(from: decoder)
+        let container = try decoder.singleValueContainer()
+        self.resumeToken = try container.decode(Document.self)
     }
 }
 
 /// A MongoDB ChangeStream.
 /// - SeeAlso: https://docs.mongodb.com/manual/changeStreams/
 public class ChangeStream<T: Codable>: Sequence, IteratorProtocol {
     /// A `ResumeToken` used for manually resuming a change stream.
```

Comment:
This may be out of scope for this review, but I think it's important to note here that this resume token is associated with the most recent event seen by this change stream. (i.e. if a user  wanted to pick up where one left off, they'd use this token).

Reply:
@patrickfreed So should `ResumeToken`'s docstring be updated or is this for its private field `resumeToken`?

Reply:
Just this field. I was saying that the docstring for this field should provide some context as to where it came from.